lib/cmd/common/build.js

```diff
@@ -1,21 +1,25 @@
 module.exports = build;
 
 build.desc = "Builds a client application";
-build.usage = "\nfhc build app=<app-id> destination=<destination> version=<version> config=<config> keypass=<private-key-password> certpass=<certificate-password> download=<true|false> provisioning=<path-to-provisioning-profile> cordova_version=<cordova version>"
+build.usage = "\nfhc build app=<app-id> destination=<destination> version=<version> config=<config> keypass=<private-key-password> certpass=<certificate-password> download=<true|false> provisioning=<path-to-provisioning-profile> cordova_version=<cordova version> branch=<branch name> commit=<commit hash>"
   + "\nwhere <destination> is one of: android, iphone, ipad, ios(for universal binary), blackberry, windowsphone7, windowsphone (windows phone 8)"
   + "\nwhere <version> is specific to the destination (e.g. Android version 4.0)"
   + "\nwhere <config> is either 'debug' (default), 'distribution', or 'release'"
   + "\nwhere <provisioning> is the path to the provisioning profile"
   + "\nwhere <cordova_version> is for specifying which version of Cordova to use. Currently supported: either 2.2 or 3.3. Only valid for Android for now."
+  + "\nwhere <branch name> is the name of the git branch to build, defaults to 'master'"
+  + "\nwhere <commit hash> is the full hash of the commit to build, if not the head of the branch. Muss be used with <branch name>"
```

Comment:
After tests on millicore I found out that specifying only the hash does not
work, `gitRef.hash` is ignored if alone. I find it weird since it shouldn't
matter from Git's perspective. The relevant code on millicore should be at
https://github.com/fheng/millicore/blob/master/src/main/java/org/tssg/millicore/scm/ScmManagerImpl.java#L865

The client's use case has to do with targeting a branch for CI builds, so
being able to target just the head of a branch is convenient, as opposed to
having to figure out the hash of the commit through a script.

I'll update the JIRA to contain this information.

On Mon, Mar 21, 2016 at 5:06 PM, wtrocki <notifications@github.com> wrote:

> In lib/cmd/common/build.js
> <https://github.com/feedhenry/fh-fhc/pull/191#discussion_r56887500>:
>
> >    + "\nwhere <destination> is one of: android, iphone, ipad, ios(for universal binary), blackberry, windowsphone7, windowsphone (windows phone 8)"
> >    + "\nwhere <version> is specific to the destination (e.g. Android version 4.0)"
> >    + "\nwhere <config> is either 'debug' (default), 'distribution', or 'release'"
> >    + "\nwhere <provisioning> is the path to the provisioning profile"
> >    + "\nwhere <cordova_version> is for specifying which version of Cordova to use. Currently supported: either 2.2 or 3.3. Only valid for Android for now."
> > +  + "\nwhere <branch name> is the name of the git branch to build, defaults to 'master'"
> > +  + "\nwhere <commit hash> is the full hash of the commit to build, if not the head of the branch. Muss be used with <branch name>"
>
> Muss => Must
>
> "Muss be used with " - I'm not sure if I'm getting it right. Do we want
> here to specify both branch and hash at the same time?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/feedhenry/fh-fhc/pull/191/files/5646381aadc032a037363bf05fdcc4c119064135#r56887500>
>